app/src/main/java/org/wikipedia/settings/SiteInfoClient.kt

```diff
@@ -0,0 +1,52 @@
+package org.wikipedia.settings
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.schedulers.Schedulers
+import org.wikipedia.Constants
+import org.wikipedia.WikipediaApp
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.dataclient.mwapi.MwQueryResponse
+import org.wikipedia.staticdata.MainPageNameData
+import org.wikipedia.util.log.L
+
+object SiteInfoClient {
+    private val SITE_INFO_MAP = mutableMapOf<String, SiteInfo?>()
+
+    @JvmStatic
+    fun getMainPageForLang(lang: String): String {
+        val info = getSiteInfoForLang(lang)
+        return if (info != null && !info.mainpage.isNullOrEmpty()) {
```

Comment:
Can we remove the null check like and us info?.mainpage... instead?

Reply:
The null check is actually necessary since we are going to use its `mainpage`. But I rewrite the code a bit.